pkg/scalers/github_runner_scaler.go

```diff
@@ -0,0 +1,632 @@
+	} `json:"license"`
+	Forks      int `json:"forks"`
+	OpenIssues int `json:"open_issues"`
+	Watchers   int `json:"watchers"`
+}
+
+type Jobs struct {
+	TotalCount int   `json:"total_count"`
+	Jobs       []Job `json:"jobs"`
+}
+
+type Job struct {
+	ID          int       `json:"id"`
+	RunID       int       `json:"run_id"`
+	RunURL      string    `json:"run_url"`
+	NodeID      string    `json:"node_id"`
+	HeadSha     string    `json:"head_sha"`
+	URL         string    `json:"url"`
+	HTMLURL     string    `json:"html_url"`
+	Status      string    `json:"status"`
+	Conclusion  string    `json:"conclusion"`
+	StartedAt   time.Time `json:"started_at"`
+	CompletedAt time.Time `json:"completed_at"`
+	Name        string    `json:"name"`
+	Steps       []struct {
+		Name        string    `json:"name"`
+		Status      string    `json:"status"`
+		Conclusion  string    `json:"conclusion"`
+		Number      int       `json:"number"`
+		StartedAt   time.Time `json:"started_at"`
+		CompletedAt time.Time `json:"completed_at"`
+	} `json:"steps"`
+	CheckRunURL     string   `json:"check_run_url"`
+	Labels          []string `json:"labels"`
+	RunnerID        int      `json:"runner_id"`
+	RunnerName      string   `json:"runner_name"`
+	RunnerGroupID   int      `json:"runner_group_id"`
+	RunnerGroupName string   `json:"runner_group_name"`
+	WorkflowName    string   `json:"workflow_name"`
+	HeadBranch      string   `json:"head_branch"`
+}
+
+// NewGitHubRunnerScaler creates a new GitHub Runner Scaler
+func NewGitHubRunnerScaler(config *ScalerConfig) (Scaler, error) {
+	httpClient := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, false)
+
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %w", err)
+	}
+
+	meta, err := parseGitHubRunnerMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing GitHub Runner metadata: %w", err)
+	}
+
+	return &githubRunnerScaler{
+		metricType: metricType,
+		metadata:   meta,
+		httpClient: httpClient,
+		logger:     InitializeLogger(config, "github_runner_scaler"),
+	}, nil
+}
+
+// getValueFromMetaOrEnv returns the value of the given key from the metadata or the environment variables
+func getValueFromMetaOrEnv(key string, metadata map[string]string, env map[string]string) (string, error) {
+	if val, ok := metadata[key]; ok && val != "" {
+		return val, nil
+	} else if val, ok := metadata[key+"FromEnv"]; ok && val != "" {
+		return env[val], nil
+	}
+	return "", fmt.Errorf("no %s given", key)
+}
+
+func getOwner(runnerScope string, config *ScalerConfig) (string, error) {
+	var err error
+	var owner string
+	switch runnerScope {
+	case ORG:
+		owner, err = getValueFromMetaOrEnv("owner", config.TriggerMetadata, config.ResolvedEnv)
+	case ENT:
+		owner, err = getValueFromMetaOrEnv("owner", config.TriggerMetadata, config.ResolvedEnv)
+	default:
+		owner, err = getValueFromMetaOrEnv("owner", config.TriggerMetadata, config.ResolvedEnv)
+	}
```

Comment:
Aren't all the cases using the same code?

Reply:
Just implementing and testing now